cmd/utils/flags.go

```diff
@@ -2156,8 +2156,29 @@ func MakeChain(ctx *cli.Context, stack *node.Node, readonly bool) (*core.BlockCh
 	}
 	vmcfg := vm.Config{EnablePreimageRecording: ctx.Bool(VMEnableDebugFlag.Name)}
 
+	// Override the chain config with provided settings.
+	var overrides core.ChainOverrides
+	if ctx.IsSet(OverrideCancun.Name) {
+		v := ctx.Uint64(OverrideCancun.Name)
+		overrides.OverrideCancun = &v
+	}
+	if ctx.IsSet(OverridePrague.Name) {
+		v := ctx.Uint64(OverridePrague.Name)
+		overrides.OverridePrague = &v
+	}
+	if ctx.IsSet(OverrideProofInBlock.Name) {
+		v := ctx.Bool(OverrideProofInBlock.Name)
+		overrides.OverrideProofInBlock = &v
+	}
+	if ctx.IsSet(OverrideOverlayStride.Name) {
+		v := ctx.Uint64(OverrideOverlayStride.Name)
+		overrides.OverrideOverlayStride = &v
+	}
+	if ctx.IsSet(ClearVerkleCosts.Name) {
+		params.ClearVerkleWitnessCosts()
+	}
```

Comment:
I'd to do this parsing here since it was passing `nil`, and `NewBlockChain(...)` implementation assumed it wasn't nil and panicked. Also, it's useful to actually override those configs.

See suggested command in the PR description.